app/templates/main/search-transferring-body.html

```diff
@@ -90,36 +91,63 @@ <h2 class="govuk-body browse__body">You are viewing</h2>
                                                 {% for record in results %}
                                                     {% set main_loop_index = loop.index %}
                                                     <tr class="govuk-table__row govuk-table__row--primary">
+                                                        {% set first_key = record['highlight'].keys() | list | first %}
+                                                        {% set first_value = record['highlight'].values() | list | first | join(' ... ')| clean_tags | safe %}
                                                         <td class="govuk-table__cell">
                                                             <div class="govuk-table__cell__data">
                                                                 <details class="details-toggle-{{ main_loop_index }}"
                                                                          {% if open_all %}open{% endif %}>
                                                                     <summary />
                                                                 </details>
                                                                 {% if record['highlight'] %}
-                                                                    {% set key = record['highlight'].keys() | list | first | replace('_', ' ') | capitalize %}
-                                                                    {{ key }} +{{ record['highlight'].keys() | length - 1 }}
+                                                                    {% set count = record['highlight'].keys() | length - 1 %}
+                                                                    {% set field = first_key | replace('_', ' ') | capitalize %}
```

Comment:
There is a lot of logic in these jinja templates now. I think we should start unit testing them directly rather than only when testing the routes as the routes themselves have a lot of logic.

I will create a ticket(or tickets) for us to start doing this across our current templates. We should also split out a lot of them as they are quite large.

something like


```
def test_template_a():
# assume it is an unittest function
    context = { 
        'test_var': 'test_value'
    }
    path = 'path/to/template/dir'
    filename = 'template_to_test.tpl'
    rendered = jinja2.Environment(
        loader=jinja2.FileSystemLoader(path)
    ).get_template(filename).render(context)
    expected = #......
    assert rendered == expected
```

Reply:
100%, this would be a much better approach